### PR TITLE
Fixes formulas for Sor

### PR DESCRIPTION
**Pull request by MarcelKoch**
This PR fixes the SOR (and Gauss-Seidel) documentation.

### CODE REVIEW

File: include/ginkgo/core/preconditioner/sor.hpp [hunk 1]
```diff
@@ -26,16 +26,16 @@ namespace preconditioner {
...
- * $$
+ * \f[
  * M = \frac{1}{\omega} (D + \omega L), \quad 0 < \omega < 2.
- * $$
+ * \f]
  * $\omega$ is known as the relaxation factor.
  * The preconditioner can be made symmetric, leading to the SSOR preconitioner.
  * Here, $M$ is defined as
- * $$
+ * \f[
```
Comment (yhmtsai) [inline include/ginkgo/core/preconditioner/sor.hpp:35]: I thought `\[` is enough, does it need to `\f[`?

Comment (MarcelKoch) [reply include/ginkgo/core/preconditioner/sor.hpp:35]: No `\[` doesn't work, it has to be `\f[`.

Comment (yhmtsai) [reply include/ginkgo/core/preconditioner/sor.hpp:35]: I assume you also look this page https://www.doxygen.nl/manual/formulas.html
maybe we or they have the tranformation from `$` to `\f$`? I wonder the transformation makes `\f[` to `\ff[`

Comment (MarcelKoch) [reply include/ginkgo/core/preconditioner/sor.hpp:35]: I was a bit confused by that, somehow `\f$` didn't work, but `$`works as expected.

Comment (yhmtsai) [reply include/ginkgo/core/preconditioner/sor.hpp:35]: I found we have used `\f[` and the formula is shown in https://ginkgo-project.github.io/ginkgo-generated-documentation/doc/develop/classgko_1_1matrix_1_1Fft.html.
I think we can rely on `\f[`

**Review by yhmtsai — APPROVED** [review]

**Pull request merged**